### DISCUSSION TITLE
How to generate a symbol from a subcircuit file?

### DISCUSSION THREAD

**Discussion by medwatt** [ra3xdh/qucs_s#359] [Q&A] [answered]
I have the following subcircuit.

```
* Subcircuit

.subckt my_capacitor p n
C1 p n 1n
.ends my_capacitor
```

Pressing F9 to create a symbol does not create ports.

![image](https://github.com/ra3xdh/qucs_s/assets/17733465/602ad004-1dda-4ec5-8f80-18457bef417a)

**Comment by ra3xdh** [comment, marked as answer]
This approach doesn't work for SPICE models. Refer to the documentation for the proper solution: https://qucs-s-help.readthedocs.io/en/latest/SubLib.html#using-manufacturers-component-data-libraries The documentation describes OpAmp example. 

There are two devices that allows to attach SPICE models:
* SPICE file
* SpiceLibComp -- for the whole library reference

The *SPICE file* device is more preferable. This device could be wrapped in a Qucs subcircuit and then library could be generated.
